x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -101,7 +104,29 @@ export const xyChartRenderer: RenderFunction<XYChartProps> = {
   },
 };
 
+function getIconForSeriesType(seriesType: SeriesType): IconType {
+  return chartTypeIcons.find(chartTypeIcon => chartTypeIcon.id === seriesType)!.iconType;
+}
+
 export function XYChart({ data, args }: XYChartProps) {
+  if (data.rows.length === 0) {
+    return (
+      <EuiFlexGroup gutterSize="s" direction="column" alignItems="center" justifyContent="center">
+        <EuiFlexItem>
+          <EuiIcon type={getIconForSeriesType(args.seriesType)} color="subdued" size="l" />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiText color="subdued" size="xs">
+            <FormattedMessage
+              id="xpack.lens.xyVisualization.noDataLabel"
+              defaultMessage="No results found"
```

Comment:
I see that there's some custom stuff here for the XY vis, but what if this was done by the frame on behalf of all visualizations?

Reply:
I think we should leave this to the visualization to be more flexible in how to show "no data". E.g. a gauge or a metric could just displayed as a dash - a map on the other hand probably just wants to show a map without data.

<img width="2515" alt="Screenshot 2019-07-22 at 09 20 47" src="https://user-images.githubusercontent.com/1508364/61613613-1627f280-ac62-11e9-981f-b1597a03b92d.png">

This is a current dashboard without any data.